npm/src/upload/index.ts

```diff
@@ -48,11 +48,12 @@ export class UploadFiles {
             throw Error("No files selected")
           }
 
-          this.clientFileProcessing.processClientFiles(
+          await this.clientFileProcessing.processClientFiles(
             consignmentId,
             files,
             progress => console.log(progress.percentageProcessed)
           )
+          location.replace(`/${consignmentId}/records`)
```

Comment:
Does using the replace function have any consequences for other behaviours? For example the "back" button? https://developer.mozilla.org/en-US/docs/Web/API/Location

Reply:
Yeah, I wasn't 100% sure on whether we wanted users to be able to click the back button, so I went for replace which doesn't allow it. I had a feeling being able to go back might add some more complexity to our form logic since we don't want multiple uploads of the same thing. Would assign() - which allows user to press back - be a better choice?

Reply:
Probably want to check the requirements around what should be happening with this

Reply:
If we do this as a form submissions, like in the prototype, does it preserve the back button behaviour?

If so, I think we should go with that for now. I've got a note to discuss back buttons with Kirsten and Sarra - both the "Back" link on the page, and the back button in the browser.

Reply:
I talked to Kirsten about the Back link and the browser's back button, and we're going to have a discussion about all the possible scenarios to decide what should happen. That discussion might not happen for a while, though.

@Lholbourn @TomJKing What do you think about changing this to a form submission, to try to preserve the back button behaviour for now?

Reply:
@suzannehamilton How big a change do you think it would be to make it a form submission?

Reply:
I don't think it's a huge change. You'll need to:

- Add a new controller endpoint which handles form submission - all it needs to do is redirect to the next page
- Update the `<form>` element in the HTML template so that form submission links to the new controller endpoint
- Capture the form submission event, and submit it when the upload is finished. Here's how we did it in the prototype: https://github.com/nationalarchives/tdr-prototype-mvc/blob/167a45dcb820777ae8d24859147fadb01ce194d2/js-src/upload/index.ts#L54

Reply:
Ah cool, I'll give it a go today!